### PR TITLE
`mod obu`: backport trailing bits check from dav1d 1.2.1

### PR DESCRIPTION
**Pull request by fbossen**

### CODE REVIEW

File: src/obu.rs [hunk 1]
```diff
@@ -122,6 +122,29 @@ impl Debug {
...
+
+    let mut size = gb.remaining_len();
+
+    while size > 0 && gb[size - 1] == 0 {
+        size -= 1;
+    }
+
+    if size != 0 {
+        return Err(EINVAL);
+    }
```
Comment (kkysen) [inline src/obu.rs:143]:
```suggestion
    if !gb.get_bytes(gb.remaining_len()).iter().all(|&b| b == 0) {
        return Err(EINVAL);
    }
```

File: src/obu.rs [hunk 2]
```diff
@@ -122,6 +122,29 @@ impl Debug {
...
+
+    return Ok(());
```
Comment (kkysen) [inline src/obu.rs:145]:
```suggestion
    Ok(())
```

**Review by kkysen — APPROVED** [review]
I made a couple of suggestions to make `fn check_trailing_bits` easier to understand/better optimized for bounds checks, but otherwise it all LGTM.

File: src/obu.rs (hunk 1 above)
Comment (fbossen) [reply src/obu.rs]: Thank you for the suggestion. Had to add a call to `gb.bytealign()` to avoid an assertion failure in `gb.get_bytes()`

**Review by kkysen — APPROVED** [review]

**Pull request merged**